app/2.0/docs/tools/polymer-json.md

```diff
@@ -0,0 +1,174 @@
+---
+title: "`polymer.json` specification"
+---
+
+<!-- toc -->
+
+# About `polymer.json`
+
+Creating a `polymer.json` in your project directory allows you to store information about your project structure and desired build configuration(s). It is used by the [Polymer CLI](sdffsdkjhk) as a way to understand the structure of your application.
+
+To make sure your application builds correctly, create a `polymer.json` file at the top level of your project:
+
+```
+-shop
+  |-README.md
+  |-app.yaml
+  |-bower.json
+  |-build
+  |-data
+  |-images
+  |-index.html
+  |-manifest.json
+  |-polymer.json
+  |-service-worker.js
+  |-src
+  |-sw-precache-config.js
+  |-test
+```
+
+#Example `polymer.json` file
+
+Here’s an example polymer.json file from the [Shop app](https://github.com/Polymer/shop):
+
+`polymer.json`
+{.caption}
+```json
+{
+  "entrypoint": "index.html",
+  "shell": "src/shop-app.html",
+  "fragments": [
+    "src/shop-list.html",
+    "src/shop-detail.html",
+    "src/shop-cart.html",
+    "src/shop-checkout.html",
+    "src/lazy-resources.html"
+  ],
+  "sources": [
+   "src/**/*",
+   "data/**/*",
+   "images/**/*",
+   "bower.json"
+  ],
+  "extraDependencies": [
+    "manifest.json",
+    "bower_components/webcomponentsjs/webcomponents-lite.min.js"
+  ],
+  "lint": {
+    "rules": ["polymer-2-hybrid"]
+  }
+}
+```
+
+## Properties
+
+### entrypoint
+Optional, Defaults to `index.html`
+Type: `String`
+
+The main entrypoint to your app for all routes, often `index.html`. This file should import the app shell file specified in the shell property. It should be as small as possible since it’s loaded and cached for all routes.
```

Comment:
It should be as small as possible because it's _not_ cached very well, since it's served for many different URLs.

Also, because it's served from many URLs, all paths should be absolute.